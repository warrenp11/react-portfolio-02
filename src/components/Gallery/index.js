import React from "react";
import { capitalizeFirstLetter } from "../../utils/helpers";
import photo from "../../assets/small/projects/0.jpg";

function Gallery() {
  const currentCategory = {
    name: "portfoltio",
    description:
      "these are a few of the projects I've created as a web developer",
  };

  return (
    <section>
      <h1>{capitalizeFirstLetter(currentCategory.name)}</h1>
      <p>{capitalizeFirstLetter(currentCategory.description)}</p>
      <div className="flex-row">
        <img
          src={photo}
          alt="Coding Quiz Thumbnail"
          className="img-thumbnail mx-1"
        />
      </div>
    </section>
  );
}

export default Gallery;
