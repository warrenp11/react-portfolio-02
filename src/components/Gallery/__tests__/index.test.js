import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Gallery from "..";

// Gallery component requires a prop of currentCategory
const Projects = {
  name: "projects",
  description: "Just a few of the projects I've created as a web developer",
};

afterEach(cleanup);

describe("Gallery is rendering", () => {
  it("renders", () => {
    render(<Gallery currentCategory={Projects} />);
  });

  // make sure the gallery matches its snapshot
  it("matches snapshot", () => {
    const { asFragment } = render(<Gallery currentCategory={Projects} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // make sure that the title of our <h1> is Projects
  it("h1 title match", () => {
    const { getByTestId } = render(<Gallery currentCategory={Projects} />);
    expect(getByTestId("h1tag")).toHaveTextContent("Projects");
  });
});