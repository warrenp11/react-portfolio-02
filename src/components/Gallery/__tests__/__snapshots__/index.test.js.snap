// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery is rendering matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Projects
    </h1>
    <p>
      Just a few of the projects I've created as a web developer
    </p>
    <div>
      <div
        class="flex-row"
      >
        <img
          alt="Coding Knowledge Quiz"
          class="img-thumbnail mx-1"
          src="0.jpg"
        />
        <img
          alt="Health Tracker App"
          class="img-thumbnail mx-1"
          src="1.jpg"
        />
        <img
          alt="Password Generator"
          class="img-thumbnail mx-1"
          src="2.jpg"
        />
        <img
          alt="Kayne West vs. Ron Swanson: Random Quote Generator"
          class="img-thumbnail mx-1"
          src="3.jpg"
        />
        <img
          alt="Work Day Scheduler"
          class="img-thumbnail mx-1"
          src="4.jpg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
